extends RayCast3D

@onready var crosshair = get_parent().get_parent().get_node("player_ui/CanvasLayer/crosshair")
@onready var player_ui = get_parent().get_parent().get_node("player_ui")

func _physics_process(_delta: float) -> void:
	if is_colliding():
		var hit = get_collider()
		if hit.name == "safe":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				player_ui.open_safe_password()
		elif hit.name == "key":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.pickup_key()
		elif hit.name == "money":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.get_parent().interact()
		elif hit.name == "light_switch":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.get_parent().toggle_light()
		elif hit.name == "door":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.get_parent().get_parent().get_parent().toggle_door()
		elif hit.name == "drawer":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.get_parent().get_parent().toggle_door()
		elif hit.name == "door_bell":
			if !crosshair.visible:
				crosshair.visible = true 
			if Input.is_action_just_pressed("interact"):
				hit.get_parent().ring_bell()
		else:
			if crosshair.visible:
				crosshair.visible = false 
	else:
		if crosshair.visible:
			crosshair.visible = false 
